File: 1. Two Sum/py/TwoSum.py
# LeetCode challenge: 1. Two Sum: <https://leetcode.com/problems/two-sum/description/>

from typing import List, Optional


class Solution:
    def twoSum(self, nums: Optional[List[int]], target: int) -> Optional[List[int]]:
        if nums is None or len(nums) < 2:
            return None
        for i in range(len(nums)):
            for j in range(i+1, len(nums)):
                if nums[i] + nums[j] == target:
                    return [i, j]

File: 1. Two Sum/cs/TwoSum/TwoSum.cs
// LeetCode challenge: 1. Two Sum: <https://leetcode.com/problems/two-sum/description/>

namespace LeetCode.TwoSum;

public class Solution
{
    public static int[]? TwoSum(int[]? nums, int target)
    {
        if (nums == null || nums.Length < 2)
            return null;
        for (int i = 0; i < nums.Length; i++)
            for (int j = i + 1; j < nums.Length; j++)
                if (nums[i] + nums[j] == target)
                    return [i, j];
        return null;
    }
}